examples/weather_bot/app/tools_base.py

```
@@ -0,0 +1,135 @@
+            summary = tool.get_summary().lower()
+            if (
+                any(word in summary for word in query_lower.split())
+                or any(kw in query_lower.split() for kw in tool.get_keywords())
+                or tool.id.lower() in query_lower
+            ):
+                results.append(tool)
+        return results[:top_k]
+
+    def load(self, tool: FunctionTool) -> FunctionTool:
+        if getattr(tool, 'id', None) not in self._tools:
+            raise ValueError(f"function tool dont have  '{tool}' defenition")
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)